includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+require_once( AMP__DIR__ . '/includes/sanitizers/class-amp-base-sanitizer.php' );
+
+/**
+ * Collects inline styles and outputs them in the amp-custom stylesheet.
+ */
+class AMP_Style_Sanitizer extends AMP_Base_Sanitizer {
+	private $styles = array();
+
+	public function sanitize() {
+		$body = $this->get_body_node();
+		$this->collect_styles_recursive( $body );
+
+		if ( ! empty( $this->styles ) ) {
+			add_action( 'amp_post_template_css', array( $this, 'append_styles' ), 99 );
```

Comment:
I initially had the same thought. I wanted to pass the data to `AMP_Post_Template` and add the action there, but I couldn't figure out a way to add data to the class without changing the access of some of the methods from private to public. Do you think it would be worth modifying `AMP_Post_Template` to make this possible?